lib/vector-common/src/lib.rs

```diff
@@ -63,3 +63,19 @@ pub type Error = Box<dyn std::error::Error + Send + Sync + 'static>;
 /// Vector's basic result type, defined in terms of [`Error`] and generic over
 /// `T`.
 pub type Result<T> = std::result::Result<T, Error>;
+
+/// Retrieve the current hostname.
```

Comment:
Seems like the comment could indicate that it is retrieved from the env var if present otherwise is computed.